### DISCUSSION TITLE
October 2022 release

### DISCUSSION THREAD

**Discussion by aucampia** [RDFLib/rdflib#2038] [Development]
The next release target date is October 2022, please use this discussion to discuss the release.

**Comment by aucampia** [comment]
I  think that it would be best to try and fix more bugs in the next release, we currently have more than [300 expected failures](https://github.com/RDFLib/rdflib/runs/7458248728?check_suite_focus=true#step:8:586), some of them are incredibly trivial to fix, if we just try fix them, and quite a lot of open issues that are bugs.

My own priorities are:
- Fix bugs
- Improve compliance testing: While the compliance testing is a lot better for some test suites, we are still not running many SPARQL tests and the JSON-LD test suite needs a lot of work
- Add type hints

All of this is aimed at improving the overall quality of RDFLib.

For interface breakages what I would like to see is:

- An [architectural decision record](https://adr.github.io/) for the new interface
- The interface made available as experimental/incubating to users so that we can see how it works before we push it out
- Interface breakages should not go half way, if we are going to break the Graph interface we should fix all problems with it in one go, because otherwise we will end up making a new major release every time we release, and ideally the starting point should be something of a survey to see what people would like changed.
- Ideally we should notify our users of planned deprecation before doing them, but I guess with major reworks this may be difficult.

Without this I'm concerned that re-architecting may end up being more of a sideways move than a clear improvement.

**Comment by aucampia** [comment]
I'm on leave next week (2022-W41), I'm going to finish whatever is needed for https://github.com/RDFLib/rdflib/issues/1844 and try and fix as many bugs as possible, and then make the release on 2022-10-23 if all goes well.

The release will be 6.3.0 as I won't include any breaking changes.

@ashleysommer @nicholascar @gjhiggins @white-gecko

**Reply by umlaeute** [reply]
ping @aucampia 

as was pointed out in https://github.com/RDFLib/rdflib/issues/1844#issuecomment-1220610012, we at Debian would like to get a fixed version rather sooner than later, as the Debian release-freeze is approaching.

**Reply by umlaeute** [reply]
ouch sorry, i wasn't aware of the ongoing discussion in #1844 where other people are already pinging like wild. i had no intention of adding additional stress :-P
